### ISSUE TITLE
Support extension updates/installation/browsing from non-Nexus sources.

### ISSUE THREAD

**Issue by agc93** [Nexus-Mods/Vortex#9527]
**Is your feature request related to a problem? Please describe.**
After recent changes in author file ownership/control, I'm trying to minimise the files I have uploaded to Nexus Mods, but currently that's the only way for extensions to be browsed in-app or (more importantly IMO) for extension updates to happen without issues.

**Describe the solution you'd like**
The way I see it there's two major categories of options: adding additional sources (i.e. GitHub or a configurable endpoint), or providing extensions an API for registering their own updates/sources so that manually installed extensions could still benefit from updates etc.

For the former, if you want to avoid dependence on a single platform, you could always define an API format like how modmeta-db is handled, or even just a variation on the `extensions_1_x.json` format that's less Nexus-specific that authors or platforms could implement. The latter would be less flexible but (at a guess) would require less implementation work.

**Additional context**

I understand this is probably a contentious addition and runs up against similar discussions in the past over the extensions update mechanism but the current situation tightly couples Vortex and Nexus Mods, even for extensions/games/developers who may not be (or want to be) on Nexus Mods.

**Comment by agc93** [comment]
I will admit I hadn't looked in the `ExtensionManager` recently and hadn't seen that GitHub support seems to already be there in some capacity, but I have no idea how extensions are populating that information in the extensions file

**Comment by TanninOne** [comment]
@agc93 We'll discuss this request internally before I can give you anything concrete.
It is true that Vortex technically is capable of fetching extensions from github but on the tooling side where we keep the extension list updated it currently requires manual work to set up a new repo and it requires the repo (releases specifically) to follow a strict pattern for updates to be picked up. This wasn't really designed with external devs in mind.

Out of curiosity: What is the technical advantage of hosting the extensions on github? I realize that you take offense in the fact that mod authors on Nexus Mods can now only archive their mods, not delete them. But on github anyone can fork your work so you can't really delete it there either?

**Comment by agc93** [comment]
@TanninOne Thanks for the update and I'd be interested to hear the result of the discussions.

Since the answer to the second point is kind of a lot, I've laid it out below.

---

This is (obviously) a personal choice, but I prefer to have greater control over distribution of **assets** (or really any non-source component), because they aren't directly controlled or covered by any code licensing I use. To keep with the GitHub example, forks are still linked to the repository and include the license and history. Conversely, forks do **not** include Releases, Wikis, Issues/PRs or anything else not covered by the license. In fact, **all** of those things are delete-able and when deleted are actually inaccessible. Of particular note given the current context, deleting a GitHub Release **will** remove the assets attached to that release, even if the underlying tag remains (since it's part of the repository itself).

To over-simplify: someone accessing/forking/clonging/anything else my source repository both a) knows the terms of its use and modification and b) is made aware that they're now interacting with **their own copy** of my work, not the original work. Distributed assets (from Releases, Nexus Mods or any other source) are either implicitly or explicitly provided as _my asset_ being made available to consumers and I don't believe that availability should be indefinite, irrevocable and unavoidable.

**Comment by agc93** [comment]
Given you highlighted that the tooling for generating the extensions file is one of the sticking points, would it be a reasonable middle ground to allow users (or extensions) to add additional endpoints for JSON responses that match the existing format used the `extensions_1_x.json` file? 

That way an author could provide their own extension source that Vortex (with suitable warnings to users about trusting third-party sources) could pull down and run through the existing extension manager logic? It would then be up to the author/publisher of how they build/manage/update that JSON source and you don't need to rework or publish internal tooling?

Just an alternative to throw into the ring.

**Comment by TanninOne** [comment]
@agc93 I have to admit I don't really follow your reasoning regarding non-code assets. Especially in the case of extensions: Archived extensions wouldn't be offered for download within Vortex and they don't get displayed on the site so users would really have to go out of their way to even get at those files, using an unofficial api client to even get at the download link. I don't think there would be any confusion about that file not being supported by you any more.

Stefan also made a news post to clarify why we're doing this in a more detailed manner: https://www.nexusmods.com/news/14538 , does that change your opinion in any way?

Now to the technical question: I don't think we have to involve user configuration in this. At the end of the day this is more of a corporate decision anyway. If we (Nexus) decide we want to allow extensions to be hosted off-site, I'd make it so they get integrated in the regular extension list and then we'd probably have a configuration file somewhere you can send a pull request for to get your extension listed and de-listed.
Yes, it would involve work on the tooling but nothing unreasonable.
If the decision is made that we don't want to allow it, adding an option for customizable endpoints would probably be undesirable for the same corporate reasons anyway.

I'd also like to clarify that this entire thing about mods only getting archived is only there to prevent collections from breaking instantly the moment one author pulls a file. This has nothing to do with extensions, you can get any extension de-listed if you want no matter how you host it.

**Comment by agc93** [comment]
That's fair enough, and that last point of clarification is also a good one I hadn't heard up until now, since extensions are a reasonably niche use case right now.

I did read the post, and since the changes to not list archived files from the API when the archive was hidden, I feel that a better middle ground has been reached with these changes. A lot of the concerns were coming from how little detail was available immediately after the change (and its inconsistent implementation), but personally I'd regard the current state as a reasonable compromise for me.

I seriously appreciate your honesty and transparency with this and can see your reasoning about the decision regarding off-site extension hosting. The only concern I would raise is that ideally any changes to extension hosting/handling/etc might be a bit better communicated than this change was

**Comment by TanninOne** [comment]
@agc93 I hope I didn't mislead you with my previous post. What I meant was that we will delist archived extensions from the extensions_?_?.json file, so that the extension won't be found through the site and the extension list in Vortex and thus to get at an extension link one would have to use a separate client. Extensions, like all mods will still be archived on the server it's just that they should be hidden pretty well.

**Comment by agc93** [comment]
Yeah that's understandable. As long as there's no chance of unsupported versions being made available in-app (impossible if its removed from the extension file) then I don't mind if the actual files are available on the site through the API, especially given the option to hide archives now being respected in the API.
